server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/chatDb/chatDb.ts

```diff
@@ -639,79 +642,123 @@ export class ChatDatabase {
             messages.pop()
             this.#features.logging.debug('Dropped trailing user message')
         }
+
+        //  Make sure there are alternating user and assistant messages
+        let i = 0
+        while (i < messages.length - 1) {
+            const currentMessageType = messages[i].type
+            const nextMessageType = messages[i + 1].type
+
+            if (currentMessageType === nextMessageType) {
+                this.#features.logging.warn(`Deleting message at index ${i + 1} to maintain alternating pattern`)
+                messages.splice(i + 1, 1)
```

Comment:
I think we should remove the last message and keep the current one

Reply:
Will update the PR